lib/widgets/share.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/scheduler.dart';
+
+import '../model/narrow.dart';
+import 'compose_box.dart';
+import 'icons.dart';
+import 'message_list.dart';
+import 'page.dart';
+import 'recent_dm_conversations.dart';
+import 'subscription_list.dart';
+
+class SharePage extends StatelessWidget {
+  const SharePage({
+    super.key,
+    required this.filesToUpload,
+    required this.sharedText,
+  });
+
+  final Iterable<FileToUpload>? filesToUpload;
+  final String? sharedText;
+
+  static AccountRoute<void> buildRoute({
+    required int accountId,
+    required Iterable<FileToUpload>? filesToUpload,
+    required String? sharedText,
+  }) {
+    return MaterialAccountWidgetRoute(
+      accountId: accountId,
+      page: SharePage(
+        filesToUpload: filesToUpload,
+        sharedText: sharedText));
+  }
+
+  void _handleNarrowSelect(BuildContext context, Narrow narrow) {
+    final composeBoxStateKey = GlobalKey<ComposeBoxState>();
+
+    unawaited(Navigator.pushReplacement(context,
+      MessageListPage.buildRoute(
+        context: context,
+        narrow: narrow,
+        composeBoxStateKey: composeBoxStateKey)));
```

Comment:
How about passing a key to use for the MessageListPage itself instead? That seems simpler to describe in the interface of MessageListPage.buildRoute.

Then for using `.currentState` to get the MessageListPageState, try this:
```diff
-abstract class MessageListPageState {
+abstract class MessageListPageState extends State<MessageListPage> {
```

and make the key local variable here have type `GlobalKey<MessageListPageState>`.